backend/json.go

```diff
@@ -69,21 +131,92 @@ func writeDataResponseJSON(dr *DataResponse, stream *jsoniter.Stream) {
 	stream.WriteObjectEnd()
 }
 
-func writeQueryDataResponseJSON(qdr *QueryDataResponse, stream *jsoniter.Stream) {
+func writeQueryDataResultsJSON(qdr *QueryDataResults, stream *jsoniter.Stream) {
 	stream.WriteObjectStart()
-	if qdr.Responses != nil {
-		stream.WriteObjectField("responses")
-		stream.WriteObjectStart()
+	if qdr.Results != nil {
+		stream.WriteObjectField("results")
+		stream.WriteArrayStart()
 		started := false
-		for id, res := range qdr.Responses {
-			if started {
-				stream.WriteMore()
+		if len(qdr.Order) > 0 {
```

Comment:
:thinking: hymmm we should likely make sure that every key actually gets used.  The order elements will match query refIds... but I don't think there are any guarantee that more keys are not added...